From 690ad847a2632290171b5acc0f15cd87a4f8cab9 Mon Sep 17 00:00:00 2001
From: Ritesh Kumar <quic_riteshk@quicinc.com>
Date: Tue, 16 Jan 2024 12:48:03 +0530
Subject: [PATCH 3/3] UPSTREAM: drm/panel: novatek-nt36672e: Include
 <linux/of.h>

Include <linux/of.h> instead of <linux/of_device.h> to fix
below compilation errors:

drivers/gpu/drm/panel/panel-novatek-nt36672e.c:564:14: error: implicit declaration of function 'of_device_get_match_data'
  ctx->desc = of_device_get_match_data(dev);
              ^
drivers/gpu/drm/panel/panel-novatek-nt36672e.c:622:34: error: array type has incomplete element type 'struct of_device_id'
 static const struct of_device_id nt36672e_of_match[] = {
                                  ^

Signed-off-by: Ritesh Kumar <quic_riteshk@quicinc.com>
Fixes: ea4f9975625a ("drm/panel: Add support for Novatek NT36672E panel driver")
Reviewed-by: Neil Armstrong <neil.armstrong@linaro.org>
Signed-off-by: Linus Walleij <linus.walleij@linaro.org>
Link: https://patchwork.freedesktop.org/patch/msgid/20240116071803.5264-1-quic_riteshk@quicinc.com
Upstream-Status: Backport [https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git 2095d5071d60fd6835f14f29621c6f234a1c0293]
---
 drivers/gpu/drm/panel/panel-novatek-nt36672e.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/panel/panel-novatek-nt36672e.c b/drivers/gpu/drm/panel/panel-novatek-nt36672e.c
index d4e85c2fc899..cb7406d74466 100644
--- a/drivers/gpu/drm/panel/panel-novatek-nt36672e.c
+++ b/drivers/gpu/drm/panel/panel-novatek-nt36672e.c
@@ -4,7 +4,7 @@
 #include <linux/delay.h>
 #include <linux/gpio/consumer.h>
 #include <linux/module.h>
-#include <linux/of_device.h>
+#include <linux/of.h>
 #include <linux/regulator/consumer.h>
 
 #include <drm/drm_mipi_dsi.h>
-- 
2.34.1

